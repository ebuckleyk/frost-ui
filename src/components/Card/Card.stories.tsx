import * as React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { BellRing, Check } from 'lucide-react';

import { cn } from '../../lib/utils';
import { Button } from '../Button';
import { Input } from '../Input';
import { Label } from '../Label';
import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '../Select';
import { Switch } from '../Switch';
import { Card, CardContent, CardDescription, CardFooter, CardHeader, CardTitle } from './Card';

const notifications = [
  {
    title: 'Your call has been confirmed.',
    description: '1 hour ago',
  },
  {
    title: 'You have a new message!',
    description: '1 hour ago',
  },
  {
    title: 'Your subscription is expiring soon!',
    description: '2 hours ago',
  },
];

type CardProps = React.ComponentProps<typeof Card>;
function CardDemo({ className, ...props }: CardProps) {
  return (
    <Card className={cn('w-[380px]', className)} {...props}>
      <CardHeader>
        <CardTitle>Notifications</CardTitle>
        <CardDescription>You have 3 unread messages.</CardDescription>
      </CardHeader>
      <CardContent className="grid gap-4">
        <div className=" flex items-center space-x-4 rounded-md border p-4">
          <BellRing />
          <div className="flex-1 space-y-1">
            <p className="text-sm font-medium leading-none">Push Notifications</p>
            <p className="text-sm text-muted-foreground">Send notifications to device.</p>
          </div>
          <Switch />
        </div>
        <div>
          {notifications.map((notification, index) => (
            <div key={index} className="mb-4 grid grid-cols-[25px_1fr] items-start pb-4 last:mb-0 last:pb-0">
              <span className="bg-sky-500 flex size-2 translate-y-1 rounded-full" />
              <div className="space-y-1">
                <p className="text-sm font-medium leading-none">{notification.title}</p>
                <p className="text-sm text-muted-foreground">{notification.description}</p>
              </div>
            </div>
          ))}
        </div>
      </CardContent>
      <CardFooter>
        <Button className="w-full">
          <Check className="mr-2 size-4" /> Mark all as read
        </Button>
      </CardFooter>
    </Card>
  );
}
function CardWithFormDemo({ className, ...props }: CardProps) {
  return (
    <Card className={cn('w-[380px]', className)} {...props}>
      <CardHeader>
        <CardTitle>Create project</CardTitle>
        <CardDescription>Deploy your new project in one-click.</CardDescription>
      </CardHeader>
      <CardContent>
        <form>
          <div className="grid w-full items-center gap-4">
            <div className="flex flex-col space-y-1.5">
              <Label htmlFor="name">Name</Label>
              <Input id="name" placeholder="Name of your project" />
            </div>
            <div className="flex flex-col space-y-1.5">
              <Label htmlFor="framework">Framework</Label>
              <Select>
                <SelectTrigger id="framework">
                  <SelectValue placeholder="Select" />
                </SelectTrigger>
                <SelectContent position="popper">
                  <SelectItem value="next">Next.js</SelectItem>
                  <SelectItem value="sveltekit">SvelteKit</SelectItem>
                  <SelectItem value="astro">Astro</SelectItem>
                  <SelectItem value="nuxt">Nuxt.js</SelectItem>
                </SelectContent>
              </Select>
            </div>
          </div>
        </form>
      </CardContent>
      <CardFooter className="flex justify-between">
        <Button variant="outline">Cancel</Button>
        <Button>Deploy</Button>
      </CardFooter>
    </Card>
  );
}

const meta: Meta<typeof Card> = {
  component: Card,
  subcomponents: { CardContent, CardDescription, CardFooter, CardHeader, CardTitle },
};

export default meta;

type Story = StoryObj<typeof Card>;

export const Demo: Story = {
  render: CardDemo,
};

export const CardWithForm: Story = {
  render: CardWithFormDemo,
};
