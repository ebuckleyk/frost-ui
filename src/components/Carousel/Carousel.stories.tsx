import * as React from 'react';
import { Meta, StoryObj } from '@storybook/react';

import { Card, CardContent } from '../Card';
import { Carousel, CarouselContent, CarouselItem, CarouselNext, CarouselPrevious } from './Carousel';

function CarouselSizeDemo() {
  return (
    <Carousel
      opts={{
        align: 'start',
      }}
      className="w-full max-w-sm"
    >
      <CarouselContent>
        {Array.from({ length: 5 }).map((_, index) => (
          <CarouselItem key={index} className="md:basis-1/2 lg:basis-1/3">
            <div className="p-1">
              <Card>
                <CardContent className="flex aspect-square items-center justify-center p-6">
                  <span className="text-3xl font-semibold">{index + 1}</span>
                </CardContent>
              </Card>
            </div>
          </CarouselItem>
        ))}
      </CarouselContent>
      <CarouselPrevious />
      <CarouselNext />
    </Carousel>
  );
}
type ComponentType = React.ComponentProps<typeof Carousel>;
const meta: Meta<ComponentType> = {
  component: Carousel,
};

export default meta;

type Story = StoryObj<ComponentType>;
export const Demo: Story = {
  render: CarouselSizeDemo,
};
