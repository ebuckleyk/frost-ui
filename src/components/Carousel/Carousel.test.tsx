import * as React from 'react';
import { render } from '@testing-library/react';

import { Card, CardContent } from '../Card';
import { Carousel, CarouselContent, CarouselItem, CarouselNext, CarouselPrevious } from './Carousel';

class IntersectionObserver {
  root = null;
  rootMargin = '';
  thresholds = [];

  disconnect() {
    return null;
  }

  observe() {
    return null;
  }

  takeRecords() {
    return [];
  }

  unobserve() {
    return null;
  }
}

window.IntersectionObserver = IntersectionObserver;
global.IntersectionObserver = IntersectionObserver;

const Component = () => {
  return (
    <Carousel
      opts={{
        align: 'start',
      }}
      className="w-full max-w-sm"
    >
      <CarouselContent>
        {Array.from({ length: 5 }).map((_, index) => (
          <CarouselItem key={index} className="md:basis-1/2 lg:basis-1/3">
            <div className="p-1">
              <Card>
                <CardContent className="flex aspect-square items-center justify-center p-6">
                  <span className="text-3xl font-semibold">{index + 1}</span>
                </CardContent>
              </Card>
            </div>
          </CarouselItem>
        ))}
      </CarouselContent>
      <CarouselPrevious />
      <CarouselNext />
    </Carousel>
  );
};

describe('Carousel', () => {
  it('should render Carousel component to match snapshot', () => {
    const result = render(<Component />);
    expect(result).toMatchSnapshot();
  });
});
