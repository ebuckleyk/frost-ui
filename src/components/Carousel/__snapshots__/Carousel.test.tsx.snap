// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Carousel > should render Carousel component to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-roledescription="carousel"
        class="relative w-full max-w-sm"
        role="region"
      >
        <div
          class="overflow-hidden"
        >
          <div
            class="flex -ml-4"
            style="transform: translate3d(NaNpx,0px,0px);"
          >
            <div
              aria-roledescription="slide"
              class="min-w-0 shrink-0 grow-0 basis-full pl-4 md:basis-1/2 lg:basis-1/3"
              role="group"
            >
              <div
                class="p-1"
              >
                <div
                  class="rounded-xl border bg-card/60 text-card-foreground shadow-md backdrop-blur-md backdrop-saturate-200"
                >
                  <div
                    class="flex aspect-square items-center justify-center p-6"
                  >
                    <span
                      class="text-3xl font-semibold"
                    >
                      1
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-roledescription="slide"
              class="min-w-0 shrink-0 grow-0 basis-full pl-4 md:basis-1/2 lg:basis-1/3"
              role="group"
            >
              <div
                class="p-1"
              >
                <div
                  class="rounded-xl border bg-card/60 text-card-foreground shadow-md backdrop-blur-md backdrop-saturate-200"
                >
                  <div
                    class="flex aspect-square items-center justify-center p-6"
                  >
                    <span
                      class="text-3xl font-semibold"
                    >
                      2
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-roledescription="slide"
              class="min-w-0 shrink-0 grow-0 basis-full pl-4 md:basis-1/2 lg:basis-1/3"
              role="group"
            >
              <div
                class="p-1"
              >
                <div
                  class="rounded-xl border bg-card/60 text-card-foreground shadow-md backdrop-blur-md backdrop-saturate-200"
                >
                  <div
                    class="flex aspect-square items-center justify-center p-6"
                  >
                    <span
                      class="text-3xl font-semibold"
                    >
                      3
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-roledescription="slide"
              class="min-w-0 shrink-0 grow-0 basis-full pl-4 md:basis-1/2 lg:basis-1/3"
              role="group"
            >
              <div
                class="p-1"
              >
                <div
                  class="rounded-xl border bg-card/60 text-card-foreground shadow-md backdrop-blur-md backdrop-saturate-200"
                >
                  <div
                    class="flex aspect-square items-center justify-center p-6"
                  >
                    <span
                      class="text-3xl font-semibold"
                    >
                      4
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-roledescription="slide"
              class="min-w-0 shrink-0 grow-0 basis-full pl-4 md:basis-1/2 lg:basis-1/3"
              role="group"
            >
              <div
                class="p-1"
              >
                <div
                  class="rounded-xl border bg-card/60 text-card-foreground shadow-md backdrop-blur-md backdrop-saturate-200"
                >
                  <div
                    class="flex aspect-square items-center justify-center p-6"
                  >
                    <span
                      class="text-3xl font-semibold"
                    >
                      5
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background backdrop-blur-md backdrop-saturate-100 transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input/10 bg-background/10 backdrop-filter-none hover:bg-accent/80 hover:text-accent-foreground absolute h-8 w-8 rounded-full -left-12 top-1/2 -translate-y-1/2"
          disabled=""
        >
          <svg
            class="h-4 w-4"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M6.85355 3.14645C7.04882 3.34171 7.04882 3.65829 6.85355 3.85355L3.70711 7H12.5C12.7761 7 13 7.22386 13 7.5C13 7.77614 12.7761 8 12.5 8H3.70711L6.85355 11.1464C7.04882 11.3417 7.04882 11.6583 6.85355 11.8536C6.65829 12.0488 6.34171 12.0488 6.14645 11.8536L2.14645 7.85355C1.95118 7.65829 1.95118 7.34171 2.14645 7.14645L6.14645 3.14645C6.34171 2.95118 6.65829 2.95118 6.85355 3.14645Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="sr-only"
          >
            Previous slide
          </span>
        </button>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background backdrop-blur-md backdrop-saturate-100 transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input/10 bg-background/10 backdrop-filter-none hover:bg-accent/80 hover:text-accent-foreground absolute h-8 w-8 rounded-full -right-12 top-1/2 -translate-y-1/2"
          disabled=""
        >
          <svg
            class="h-4 w-4"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8.14645 3.14645C8.34171 2.95118 8.65829 2.95118 8.85355 3.14645L12.8536 7.14645C13.0488 7.34171 13.0488 7.65829 12.8536 7.85355L8.85355 11.8536C8.65829 12.0488 8.34171 12.0488 8.14645 11.8536C7.95118 11.6583 7.95118 11.3417 8.14645 11.1464L11.2929 8H2.5C2.22386 8 2 7.77614 2 7.5C2 7.22386 2.22386 7 2.5 7H11.2929L8.14645 3.85355C7.95118 3.65829 7.95118 3.34171 8.14645 3.14645Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="sr-only"
          >
            Next slide
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-roledescription="carousel"
      class="relative w-full max-w-sm"
      role="region"
    >
      <div
        class="overflow-hidden"
      >
        <div
          class="flex -ml-4"
          style="transform: translate3d(NaNpx,0px,0px);"
        >
          <div
            aria-roledescription="slide"
            class="min-w-0 shrink-0 grow-0 basis-full pl-4 md:basis-1/2 lg:basis-1/3"
            role="group"
          >
            <div
              class="p-1"
            >
              <div
                class="rounded-xl border bg-card/60 text-card-foreground shadow-md backdrop-blur-md backdrop-saturate-200"
              >
                <div
                  class="flex aspect-square items-center justify-center p-6"
                >
                  <span
                    class="text-3xl font-semibold"
                  >
                    1
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-roledescription="slide"
            class="min-w-0 shrink-0 grow-0 basis-full pl-4 md:basis-1/2 lg:basis-1/3"
            role="group"
          >
            <div
              class="p-1"
            >
              <div
                class="rounded-xl border bg-card/60 text-card-foreground shadow-md backdrop-blur-md backdrop-saturate-200"
              >
                <div
                  class="flex aspect-square items-center justify-center p-6"
                >
                  <span
                    class="text-3xl font-semibold"
                  >
                    2
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-roledescription="slide"
            class="min-w-0 shrink-0 grow-0 basis-full pl-4 md:basis-1/2 lg:basis-1/3"
            role="group"
          >
            <div
              class="p-1"
            >
              <div
                class="rounded-xl border bg-card/60 text-card-foreground shadow-md backdrop-blur-md backdrop-saturate-200"
              >
                <div
                  class="flex aspect-square items-center justify-center p-6"
                >
                  <span
                    class="text-3xl font-semibold"
                  >
                    3
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-roledescription="slide"
            class="min-w-0 shrink-0 grow-0 basis-full pl-4 md:basis-1/2 lg:basis-1/3"
            role="group"
          >
            <div
              class="p-1"
            >
              <div
                class="rounded-xl border bg-card/60 text-card-foreground shadow-md backdrop-blur-md backdrop-saturate-200"
              >
                <div
                  class="flex aspect-square items-center justify-center p-6"
                >
                  <span
                    class="text-3xl font-semibold"
                  >
                    4
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-roledescription="slide"
            class="min-w-0 shrink-0 grow-0 basis-full pl-4 md:basis-1/2 lg:basis-1/3"
            role="group"
          >
            <div
              class="p-1"
            >
              <div
                class="rounded-xl border bg-card/60 text-card-foreground shadow-md backdrop-blur-md backdrop-saturate-200"
              >
                <div
                  class="flex aspect-square items-center justify-center p-6"
                >
                  <span
                    class="text-3xl font-semibold"
                  >
                    5
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background backdrop-blur-md backdrop-saturate-100 transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input/10 bg-background/10 backdrop-filter-none hover:bg-accent/80 hover:text-accent-foreground absolute h-8 w-8 rounded-full -left-12 top-1/2 -translate-y-1/2"
        disabled=""
      >
        <svg
          class="h-4 w-4"
          fill="none"
          height="15"
          viewBox="0 0 15 15"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M6.85355 3.14645C7.04882 3.34171 7.04882 3.65829 6.85355 3.85355L3.70711 7H12.5C12.7761 7 13 7.22386 13 7.5C13 7.77614 12.7761 8 12.5 8H3.70711L6.85355 11.1464C7.04882 11.3417 7.04882 11.6583 6.85355 11.8536C6.65829 12.0488 6.34171 12.0488 6.14645 11.8536L2.14645 7.85355C1.95118 7.65829 1.95118 7.34171 2.14645 7.14645L6.14645 3.14645C6.34171 2.95118 6.65829 2.95118 6.85355 3.14645Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="sr-only"
        >
          Previous slide
        </span>
      </button>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background backdrop-blur-md backdrop-saturate-100 transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input/10 bg-background/10 backdrop-filter-none hover:bg-accent/80 hover:text-accent-foreground absolute h-8 w-8 rounded-full -right-12 top-1/2 -translate-y-1/2"
        disabled=""
      >
        <svg
          class="h-4 w-4"
          fill="none"
          height="15"
          viewBox="0 0 15 15"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8.14645 3.14645C8.34171 2.95118 8.65829 2.95118 8.85355 3.14645L12.8536 7.14645C13.0488 7.34171 13.0488 7.65829 12.8536 7.85355L8.85355 11.8536C8.65829 12.0488 8.34171 12.0488 8.14645 11.8536C7.95118 11.6583 7.95118 11.3417 8.14645 11.1464L11.2929 8H2.5C2.22386 8 2 7.77614 2 7.5C2 7.22386 2.22386 7 2.5 7H11.2929L8.14645 3.85355C7.95118 3.65829 7.95118 3.34171 8.14645 3.14645Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="sr-only"
        >
          Next slide
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
